common.mk

```diff
@@ -134,9 +134,15 @@ $$($(1)_OBJ_DIR)%.o: $$($(1)_REAL_ROOT)%.S $$($(1)_MAKEFILE) $(wildcard $(ROOT)*
 	$$($(1)_CC_BASE) -c $$< -o $$@
 	$$($(1)_CC_BASE) -MM -MT $$@ -MF $$(@:.o=.d) $$<
 
-# the component is shown to depend on both obj and source files so we get a meaningful error message
-# for missing explicitly named source files
-$$($(1)_AR): $$($(1)_OBJ_FILES) $$($(1)_SRC_FILES)
+$(1)_AR_IN_FILES = $$($(1)_OBJ_FILES)
+
+# the component is shown to depend on both obj and source files so we get 
+# a meaningful error message for missing explicitly named source files
+ifeq ($(INCLUDE_SRC_IN_AR),1)
+   $(1)_AR_IN_FILES += $$($(1)_SRC_FILES)
```

Comment:
If source files is added into an archive it can not be included into a binary with option --whole-archive